packages/autorest.typescript/src/restLevelClient/transforms/transformOptions.ts

```diff
@@ -17,9 +17,9 @@ export function transformOptions(model: CodeModel): RLCOptions {
     dependencyInfo,
     productDocLink,
     azureArm,
-    flavor,
+    flavor
   } = getAutorestOptions();
-  const options: RLCOptions = {};
+  const options: RLCOptions = { moduleKind: "cjs" };
```

Comment:
I'd like to clarify some details, there are two main changes
- changed to esm 
- upgrade our build/test dependencies and steps

Are the two changes coupled with each other? Will the HLC need to adopt this? I mean if HLC won't can old code run correctly without changes?